Porting/updateAUTHORS.pl

```diff
@@ -149,6 +191,94 @@ sub main {
 
 exit(main()) unless caller;
 
+sub error_advice_for_uncommitted_changes {
+    my ($author_name, $author_email, $committer_name, $committer_email)= @_;
+    my $props= "";
+    my $extra= "";
+    my @git_env_keys=  map {
```

Comment:
I added /some/ verbiage, but i am not sure it is really necessary. Anyone tracing the code flow there would know what is up. Regardless I worked this a fair bit.

```
not ok 6 - Uncommitted changes are by a known contributor?
#   Failed test 'Uncommitted changes are by a known contributor?'
#   at Porting/updateAUTHORS.pl line 133.
#          got: '0'
#     expected: '1'
# 
# There are uncommitted changes in the working directory and your git
# credentials are new to us. We think that git thinks your credentials
# are as follows (git may use defaults we don't guess properly):
# 
#     Author Name     = bob
#     Author Email    = demerphq@gmail.com
#     Committer Name  = Yves Orton
#     Committer Email = demerphq@gmail.com
# 
# You have configured the following relevant git config settings:
# 
#     author.name      = bob
#     user.name        = Yves Orton
#     user.email       = demerphq@gmail.com
# 
# 
# Your committer and author details differ. You may want to review your
# git configuration.
# 
# 
# To resolve this you can perform one or more of these steps:
# 
#     1. Remove the uncommitted changes, including untracked files that
#        show up in
# 
#             git status
# 
#        if you wish to REMOVE UNTRACKED FILES and DELETE ANY CHANGES
#        you can
# 
#             git clean -dfx
#             git checkout -f
# 
#         BE WARNED: THIS MAY LOSE DATA.
# 
#     2. You are already configured in git and you just need to add
#        yourself to AUTHORS and other infra: commit the changes in the
#        working directory, including any untracked files that you plan to
#        add (the rest should be removed), and then run
# 
#             Porting/updateAUTHORS.pl
# 
#        to update the AUTHORS and .mailmap files automatically. Inspect
#        the changes it makes and then commit them once you are
#        satisfied. This is your option to decide who you will be known
#        as in the future!
# 
#     3. You are already a contributor to the project but you are committing
#        changes on behalf of someone who is new. Run
# 
#             Porting/updateAUTHORS.pl
# 
#        to update the AUTHORS and .mailmap files automatically. Inspect
#        the changes it makes and then commit them once you are satisfied.
#        Make sure the conributor is ok with the decisions you make before
#        you merge.
# 
#     3. You are already an author but your git config is broken or
#        different from what you expect, or you are a new author but you
#        havent configured your git details properly, in which case you
#        can use something like the following commands:
# 
#             git config --set user.name "Some Name"
#             git config --set user.email "somewhere@provider"
# 
#        If you are known to the project already this is all you need to
#        do. If you are not then you should perform option 2 or 4 as well
#        afterwards.
# 
#     4. You do not want to be listed in AUTHORS: commit the changes,
#        including any untracked unignored files, and then run
# 
#             Porting/updateAUTHORS.pl --exclude
# 
#        and commit the changes it creates. This test should pass once
#        those commits are created. Thank you for your contributions.
```